src/System.Resources.Writer/src/System/Resources/ResourceWriter.cs

```diff
@@ -268,6 +462,235 @@ private static void Write7BitEncodedInt(BinaryWriter store, int value)
+                    typeName = typeName.Substring(17);  // Remove through '.'
+                    ResourceTypeCode typeCode = (ResourceTypeCode)Enum.Parse(typeof(ResourceTypeCode), typeName);
+                    return typeCode;
+                }
+            }
+            else 
+            {
+                typeName = MultitargetingHelpers.GetAssemblyQualifiedName(type, typeConverter);
+            }
+ 
+            int typeIndex = types.IndexOf(typeName);
+            if (typeIndex == -1) {
+                typeIndex = types.Count;
+                types.Add(typeName);
+            }
+ 
+            return (ResourceTypeCode)(typeIndex + ResourceTypeCode.StartOfUserTypes);
+        }
+
+        private void WriteValue(ResourceTypeCode typeCode, Object value, BinaryWriter writer, IFormatter objFormatter)
+        {
+            Contract.Requires(writer != null);
+
+            switch (typeCode)
+            {
+                case ResourceTypeCode.Null:
+                    break;
+
+                case ResourceTypeCode.String:
+                    writer.Write((String)value);
+                    break;
+
+                case ResourceTypeCode.Boolean:
+                    writer.Write((bool)value);
+                    break;
+
+                case ResourceTypeCode.Char:
+                    writer.Write((UInt16)(char)value);
+                    break;
+
+                case ResourceTypeCode.Byte:
+                    writer.Write((byte)value);
+                    break;
+
+                case ResourceTypeCode.SByte:
+                    writer.Write((sbyte)value);
+                    break;
+
+                case ResourceTypeCode.Int16:
+                    writer.Write((Int16)value);
+                    break;
+
+                case ResourceTypeCode.UInt16:
+                    writer.Write((UInt16)value);
+                    break;
+
+                case ResourceTypeCode.Int32:
+                    writer.Write((Int32)value);
+                    break;
+
+                case ResourceTypeCode.UInt32:
+                    writer.Write((UInt32)value);
+                    break;
+
+                case ResourceTypeCode.Int64:
+                    writer.Write((Int64)value);
+                    break;
+
+                case ResourceTypeCode.UInt64:
+                    writer.Write((UInt64)value);
+                    break;
+
+                case ResourceTypeCode.Single:
+                    writer.Write((Single)value);
+                    break;
+
+                case ResourceTypeCode.Double:
+                    writer.Write((Double)value);
+                    break;
+
+                case ResourceTypeCode.Decimal:
+                    writer.Write((Decimal)value);
+                    break;
+
+                case ResourceTypeCode.DateTime:
+                    // Use DateTime's ToBinary & FromBinary.
+                    Int64 data = ((DateTime)value).ToBinary();
```

Comment:
Nit: `Int64` -> `long`. Elsewhere `Int32` -> `int`. The corefx coding style is to use language keywords instead of BCL types.